### DISCUSSION TITLE
Peer constantly added and removed

### DISCUSSION THREAD

**Discussion by SvenRuin** [markqvist/Reticulum#627] [Help & Questions]
I have followed the example of Alice and Bob on https://unsigned.io/rnode_bootstrap_console/guides/loracomms.html. The communication works, but on "Bob's" side (after rnsd -vvv) it sometimes appears like some peer is constantly added and removed, for no obvious reason. I have tried to add a photo of the screen of the Raspberry Pi. Don't know if this is expected behaviour or if I have found a bug.  
![20241203](https://github.com/user-attachments/assets/68dc0250-3d67-4c1f-93b0-7d33b51ba663)

**Comment by SvenRuin** [comment]
After looking more into this, I have realized that the peer which is added and removed must be Sideband (v1.1.3) running on a mobile phone with Sailfish OS and Android AppSupport. I can confirm that Sideband does work on that platform, but shortly after the screenlock is activated on the phone, the Reticulum gateway (on a Raspberry Pi running 64-bit OS) seems to remove this peer. Same thing happens if I shut down Sideband.

The only other problems I have noticed with Sideband on Sailfish is that:
- Under Preferences it was difficult to edit the Display Name, so although I added part of my name, I could at the end not remove Anonymous.
- When writing a message, I have to change to lower case for every letter I write, if I don't want them to be all upper case.
- It is sometimes not possible to use backspace to delete writing mistakes.

By the way, Reticulum (v0.8.6) was on the Pi installed with the command
pipx install rns
and the path was added with
pipx ensurepath

**Reply by markqvist** [reply]
Thanks for the info @SvenRuin!

Is this on a PinePhone device, by any chance? A while back, I tried getting RNS and Sideband running well on PinePhone, under a variety of different operating systems, but unfortunately never managed to, because of the power management in place on the devices.

As you have experienced, very shortly after the screen is turned off, the WiFi hardware is also put into sleep mode, and will stop outgoing and incoming packets. This of course means that the multicast peering packets that the RNS `AutoInterface` uses for peer discovery stops, and any other peers on the WiFi/Ethernet medium will automatically unpeer with the device after a short time.

On "normal" computers, this problem is non-existent, as the WiFi hardware is kept awake as a matter of default behaviour.

On Android, this is easily handled by asking the Android OS to acquire a WiFi-hardware multicast lock, which keeps the card in a state that can keep receiving data.

When I initially looked at it, I never found a way to do this on any of the operating systems that can run on PinePhone hardware, and I had to drop the project at that point :(

I don't know if anything has changed since then, or if anyone more knowledgeable than me knows of a way to keep the hardware awake (enough) to handle packets while the screen is off.

**Reply by SvenRuin** [reply]
Thank you for the reply. So the problem shown on my photo is apparently caused by WiFi going in and out of sleep mode on the peer device running Sideband (and thus not a bug in Reticulum). I think that being caused by "screen is off", as you write, could be more accurate than "screen lock". My phone is a Sony Xperia 10 II dual-SIM. Can also mention that after sending a message from it, I have once experienced on it that Sideband reports the message State as Failed (instead of Delivered), despite that the message was somehow received by "Alice".

**Reply by markqvist** [reply]
> after sending a message from it, I have once experienced on it that Sideband reports the message State as Failed (instead of Delivered), despite that the message was somehow received by "Alice".

This would be because the proof packet, that verifies delivery, was never actually received by your phone, but the message actually did get through. Again, this points to the WiFi hardware performing sub-optimally.

In situations like this, you could try resorting to using the TCP interface only. It may work better even in the sleep situations. The AutoInterface does require networking hardware that fully supports multicast to work properly, and unfortunately that seems to not be the case here.

**Reply by SvenRuin** [reply]
On Sailfish OS Forum, someone else suggested to check the recommendation on: https://forum.sailfishos.org/t/xperia-x-bluetooth-watch-problem/4874
